### DISCUSSION TITLE
Code for dijkstra's algorithm

### DISCUSSION THREAD

**Discussion by TheDeathOMG** [networkx/networkx#6084] [Algorithms]
A month ago I started a discussion like this, however I do not know how can I see the code that uses this library in:
  - dijkstra_path(G, source, target, weight='weight')

Someone told me to go to: networkx/networkx/algorithms/shortest_paths/weighted.py, but i do not find where the code is and I only see how can you use it to work the function. 

I only want this to improve the explanation of the code of Dijkstra's algorithm because with heapq I have problems with the lists of networkx. Thank you for reading!

**Comment by MridulS** [comment]
The code is available at https://github.com/networkx/networkx/blob/bcf607cf7ce4009ca37786b2fcd84e548f1833f5/networkx/algorithms/shortest_paths/weighted.py#L81, if you scroll down you will see that this function calls https://github.com/networkx/networkx/blob/bcf607cf7ce4009ca37786b2fcd84e548f1833f5/networkx/algorithms/shortest_paths/weighted.py#L376, which calls https://github.com/networkx/networkx/blob/bcf607cf7ce4009ca37786b2fcd84e548f1833f5/networkx/algorithms/shortest_paths/weighted.py#L631, which further calls https://github.com/networkx/networkx/blob/bcf607cf7ce4009ca37786b2fcd84e548f1833f5/networkx/algorithms/shortest_paths/weighted.py#L764.

And the last function `_dijkstra_multisource` is the place where the code is written down which is used in `dijkstra_path`.
